If you're running samba-1.9.17 or newer, you probably don't need this
patch.  It is required for versions up to 1.9.16p11, though.

diff -cr samba-1.9.16p2/source/client.c samba-1.9.16p2.1/source/client.c
*** samba-1.9.16p2/source/client.c	Fri Aug 23 01:39:51 1996
--- samba-1.9.16p2.1/source/client.c	Mon Sep 16 12:41:10 1996
***************
*** 85,90 ****
--- 85,91 ----
  		     int *rprcnt,int *rdrcnt,
  		     char *param,char *data,
  		     char **rparam,char **rdata);
+ static BOOL do_this_one(file_info *finfo);
  
  
  /* clitar bits insert */
***************
*** 135,140 ****
--- 136,143 ----
  int put_total_size = 0;
  int put_total_time_ms = 0;
  
+ /* totals globals */
+ int dir_total = 0;
  
  extern int Client;
  
***************
*** 473,484 ****
    ****************************************************************************/
  static void display_finfo(file_info *finfo)
  {
!   time_t t = finfo->mtime; /* the time is assumed to be passed as GMT */
!   DEBUG(0,("  %-30s%7.7s%10d  %s",
! 	   CNV_LANG(finfo->name),
! 	   attrib_string(finfo->mode),
! 	   finfo->size,
! 	   asctime(LocalTime(&t))));
  }
  
  
--- 476,491 ----
    ****************************************************************************/
  static void display_finfo(file_info *finfo)
  {
!   if (do_this_one(finfo)) {
!     time_t t = finfo->mtime; /* the time is assumed to be passed as GMT */
!     DEBUG(0,("  %-30s%7.7s%10d  %s",
! 	     CNV_LANG(finfo->name),
! 	     attrib_string(finfo->mode),
! 	     finfo->size,
! 	     asctime(LocalTime(&t))));
! 
!     dir_total += finfo->size;
!   }
  }
  
  
***************
*** 1104,1109 ****
--- 1111,1117 ----
    fstring buf;
    char *p=buf;
  
+   dir_total = 0;
    strcpy(mask,cur_dir);
    if(mask[strlen(mask)-1]!='\\')
      strcat(mask,"\\");
***************
*** 1122,1127 ****
--- 1130,1137 ----
    do_dir(inbuf,outbuf,mask,attribute,NULL,recurse);
  
    do_dskattr();
+ 
+   fprintf(stderr, "Total bytes listed: %d\n", dir_total);
  }
  
  
diff -cr samba-1.9.16p2/source/clitar.c samba-1.9.16p2.1/source/clitar.c
*** samba-1.9.16p2/source/clitar.c	Mon Jun 10 14:49:10 1996
--- samba-1.9.16p2.1/source/clitar.c	Mon Sep  9 17:10:31 1996
***************
*** 41,47 ****
  #endif
  
  static char *tarbuf;
! static int tp, ntarf, tbufsiz;
  /* Incremental mode */
  BOOL tar_inc=False;
  /* Reset archive bit */
--- 41,47 ----
  #endif
  
  static char *tarbuf;
! static int tp, ntarf, tbufsiz, ttarf;
  /* Incremental mode */
  BOOL tar_inc=False;
  /* Reset archive bit */
***************
*** 271,278 ****
    tbufsiz=blocksize*TBLOCK;
    tarbuf=malloc(tbufsiz);
  
!   /* reset tar buffer pointer and tar file counter */
!   tp=0; ntarf=0;
  }
  
  /****************************************************************************
--- 271,278 ----
    tbufsiz=blocksize*TBLOCK;
    tarbuf=malloc(tbufsiz);
  
!   /* reset tar buffer pointer and tar file counter and total dumped */
!   tp=0; ntarf=0; ttarf=0;
  }
  
  /****************************************************************************
***************
*** 1038,1043 ****
--- 1038,1044 ----
        if (finfo.size % TBLOCK)
  	dozerobuf(tarhandle, TBLOCK - (finfo.size % TBLOCK));
        
+       ttarf+=finfo.size + TBLOCK - (finfo.size % TBLOCK);
        ntarf++;
      }
    
***************
*** 1571,1576 ****
--- 1572,1578 ----
      free(tarbuf);
      
      DEBUG(0, ("tar: dumped %d tar files\n", ntarf));
+     fprintf(stderr, "Total bytes written: %d\n", ttarf);
      break;
    }
  

